src/state/apis/covalent/validators.ts

```diff
@@ -215,32 +214,44 @@ export const parseToV2NftUserItems = (
           ]
         : []
 
-      const token: V2ZapperNft = {
-        id: covalentItem.contract_address,
+      const item: NftItem = {
         name: covalentItem.contract_name,
-        tokenId: nftData.token_id,
-        lastSaleEth: null, // Covalent doesn't provide this information
-        rarityRank: null, // Covalent doesn't provide this information
-        estimatedValueEth: null, // Map this property if available in CovalentNftItemSchemaType
+        chainId,
+        id: nftData.token_id,
         medias,
+        price: null, // Covalent doesn't provide spot pricing for NFT items
+        rarityRank: null, // Covalent doesn't provide rarity rank
         collection: {
-          address: covalentItem.contract_address,
-          network: chainIdToZapperNetwork(chainId),
-          name: covalentItem.contract_name,
-          nftStandard: covalentItem.type,
-          type: 'general',
-          floorPriceEth: null,
-          logoImageUrl: covalentItem.logo_url,
-          openseaId: null, // Doesn't exist in Covalent API
+          id:
+            covalentItem.contract_address && covalentItem.supports_erc?.length
+              ? toAssetId({
+                  // Yeah, it's weird, but this is how Covalent does it
+                  assetReference: covalentItem.contract_address,
+                  assetNamespace: covalentItem.supports_erc[
+                    covalentItem.supports_erc.length - 1
+                  ] as AssetNamespace,
+                  chainId,
+                })
+              : '',
+          chainId,
+          name: covalentItem.contract_name ?? 'Collection',
+          description: nftData.external_data?.description || '',
```

Comment:
```suggestion
          description: nftData.external_data?.description ?? '',
```

Reply:
Might be an empty string or undefined, so we need to actually `||` it or that will only work in case `description` is null